apps/api/src/app/workflows-v2/usecases/get-workflow/get-workflow.usecase.ts

```diff
@@ -40,6 +38,20 @@ export class GetWorkflowUseCase {
     return toResponseWorkflowDto(notificationTemplateEntity, preferences, stepIdToControlValuesMap);
   }
 
+  private async findById(command: GetWorkflowCommand) {
+    try {
+      return await this.notificationTemplateRepository.findByIdQuery({
+        id: command._workflowId,
+        environmentId: command.user.environmentId,
+      });
+    } catch (err) {
+      if (err instanceof Error.CastError) {
+        throw new WorkflowNotFoundException(command._workflowId);
+      }
+      throw err;
```

Comment:
I'm not sure if, in this layer, we should use Mogo specific errors, we should catch that error in the repository and throw the `DalException` instead and